Add Chef Infra Client 17.1 release notes (#11510)
chef/chef: RELEASE_NOTES.md

We've updated Chef InSpec from 4.33.1 to 4.36.4:

- Added the selinux resource which includes support for modules and booleans.
- Added the pattern input option for DSL and metadata inputs.
- Added the `members_array` property for group & groups resources.
- Train now reads the username and port from the `.ssh/config` file and will use these values if present.
- Switch to GNU timeout-based implementation of SSH timeouts.
- Fixed the group resource when a member does not exist.

### Unified Mode Improvements

We've extended support for Unified Mode to the `edit_resource` helper and also improved the Unified Mode related deprecation warnings to provide more useful information and not warn when resources are deprecated or set to only run on older Chef Infra Client releases.

### Resource Improvements

#### hostname

The `hostname` resource now supports changing the hostname on Active Directory connected Windows hosts when using new `domain_user` and `domain_password` properties.

#### service on systemd Hosts

The `service` resource, on systemd hosts, will now properly load the state of the service. Thanks for this fix [@ramereth](https://github.com/ramereth)!

#### systemd_unit

> The `systemd_unit` resource has been updated to resolve a regression in Chef Infra Client 17.0 that would re-enable and restart unit files on each Chef Infra Client run. Thanks for this fix [@gene1wood](https://github.com/gene1wood)!

Reviewer: ```suggestion
We updated the `systemd_unit` resource to resolve a regression in Chef Infra Client 17.0 that would re-enable and restart unit files on each Chef Infra Client run. Thanks for this fix [@gene1wood](https://github.com/gene1wood)!
```

#### template

> The `template` resource has been updated to allow passing the `cookbook_name` variable to template files.

Reviewer: ```suggestion
We updated the `template` resource to allow passing the `cookbook_name` variable to template files.
```

#### Windows Resource

We fixed a failure that could occur in multiple Windows resources due to larger 64bit values that logged the error: `RangeError: bignum too big to convert into 'long'`.

#### windows_security_policy

The `windows_security_policy` resource now supports setting `AuditPolicyChange` and `LockoutDuration`.

#### yum_package / dnf_package

> We've made multiple improvements to how we interact with the systems RPM database in the `yum_package` and `dnf_package` resources. These changes improve reliability interacting with the rpm database and includes significant performance improvements, especially when no installation or upgrade action is taken by Chef Infra Client.

Reviewer: ```suggestion
We've made multiple improvements to how we interact with the systems RPM database in the `yum_package` and `dnf_package` resources. These changes improve reliability interacting with the RPM database and includes significant performance improvements, especially when no installation or upgrade action is taken by Chef Infra Client.
```

### Platform Detection

[Rocky Linux](https://rockylinux.org/), a RHEL clone, is now detected as a member of the `rhel` platform family.

### Packaging

> ### Improved Depedencies

Reviewer: ```suggestion
### Improved Dependencies
```

Chef Infra Client 17.1 is once again smaller than previous releases thanks to reduced dependencies in the packages.

### RHEL 8 Packages

> Our RHEL 8 packages have been improved with additional RHEL 8 optimizations and EL8 in the filename.

Reviewer: ```suggestion
We imporved our RHEL 8 packages with additional RHEL 8 optimizations and EL8 in the filename.
```
